corehq/apps/geospatial/es.py

```diff
@@ -131,3 +134,34 @@ def mid(lower, upper):
     """
     assert lower <= upper
     return ceil(lower + (upper - lower) / 2)
+
+
+def case_query_for_missing_geopoint_val(domain, geo_case_property, case_type=None, size=None, offset=0):
+    query = (
+        CaseSearchES()
+        .domain(domain)
+        .filter(_geopoint_value_missing_for_property(geo_case_property))
+    )
+    if case_type:
+        query = query.case_type(case_type)
+    query.sort('opened_on')
```

Comment:
I'd suggest sorting only when needed.
There might be a certain advantage if you could skip sorting when just the count is needed.
Same for start.

Reply:
We're not refreshing the index anymore after each batch run. Once a batch of cases have been processed, it might take a few seconds for these changes to reflect in ES. The sorting here is used as a way of paginating through the query in a predictable manner so that we don't accidentally fetch cases that have already been processed but aren't reflected yet in ES.

@mkangia I'm open to other ideas if you think there might be a better way in which we can ensure we don't process the same cases again.

Reply:
Are there any use cases where you don't need the pagination but just a one shot straight count of records?

Reply:
The only time this would happen is if the total case count is less than `DEFAULT_QUERY_LIMIT` (10k cases). I'm not sure how likely this will be, but I can definitely see this being a valid edge case.

I'll do some refactoring to make sorting and offset optional for the above scenario.

Reply:
Addressed in [1de4f92](https://github.com/dimagi/commcare-hq/pull/35126/commits/1de4f927ad98c2b1eb96ed16d4c9728f5e9ec79a).